malaria-app-android/src/main/java/com/peacecorps/malaria/ui/user_profile/UserProfileFragment.java

```diff
@@ -0,0 +1,193 @@
+package com.peacecorps.malaria.ui.user_profile;
+
+/*
+ * Created by Anamika Tripathi on 13/7/18.
+ */
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.peacecorps.malaria.R;
+import com.peacecorps.malaria.ui.base.BaseFragment;
+import com.peacecorps.malaria.utils.InjectionClass;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class UserProfileFragment extends BaseFragment implements UserProfileContract.UserProfileMvpView {
+
+    private OnUserFragmentListener mListener;
+    private Context context;
+    //Butterknife variables
+    @BindView(R.id.edit_text_user_name)
+    EditText userName;
+    @BindView(R.id.edit_text_user_email)
+    EditText userEmail;
+    @BindView(R.id.edit_text_user_age)
+    EditText userAge;
+    @BindView(R.id.edit_text_user_medicine_type)
+    EditText userMedicationType;
+
+    private UserProfilePresenter<UserProfileFragment> presenter;
+
+    public static UserProfileFragment newInstance() {
+//        Bundle args = new Bundle();
+//        args.putString(ARG_PARAM1, param1);
+//        args.putString(ARG_PARAM2, param2);
+//        fragment.setArguments(args);
+        return new UserProfileFragment();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view =  inflater.inflate(R.layout.user_profile, container, false);
+        context = getContext();
+        if(getActivity()!=null)
+            ButterKnife.bind(this, view);
+        presenter = new UserProfilePresenter<>(InjectionClass.provideDataManager(context), context);
+        presenter.attachView(this);
+        return view;
+    }
+
+    @Override
+    protected int getContentResource() {
+        return R.layout.user_profile;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        init();
+    }
+
+    @Override
+    protected void init() {
+        presenter.setPreviousDetails();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof OnUserFragmentListener) {
+            mListener = (OnUserFragmentListener) context;
+        } else {
+            throw new RuntimeException(context.toString()
```

Comment:
as onAttach, is public function, the exception will be thrown back to caller, and you should have right declaration to inform caller to handle it, otherwise it will crash your application.